### DISCUSSION TITLE
Add current step preview on recursion steps

### DISCUSSION THREAD

**Discussion by KailasMahavarkar** [pamelafox/recursive-visualizations#24] [Ideas]
My sample code:

```
from typing import List

def solve(answer: List[List[int]], nums: List[int], idx: int):
    if idx == len(nums):
        answer.append(nums[:])
        return -1

    for i in range(idx, len(nums)):
        nums[idx], nums[i] = nums[i], nums[idx]
        solve(answer, nums, idx + 1)
        nums[idx], nums[i] = nums[i], nums[idx]
    return -1

class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        answer = []
        solve(answer, nums, 0)
        return answer
```

It would be a good idea to add recursion step on arrows (or there should be a button to toggle it)

![image](https://github.com/pamelafox/recursive-visualizations/assets/66670953/98074bf3-b4ec-41f5-b7cb-a0416f3e6bb4)

**Comment by pamelafox** [comment]
Nice idea! I've also had the suggestion to use color coding, like a gradient, to indicate the progression. I don't have much free time at the moment, but pull requests are welcome if you want to tinker with it.

**Reply by KailasMahavarkar** [reply]
Sure, let me see what i can do ... maybe i'll work on this feature
